build_rules.gradle

```diff
@@ -593,6 +593,118 @@ ext.applyJavaNature = {
+
+  String integrationTestPipelineOptions = System.getProperty('integrationTestPipelineOptions')
+
+  // PerfKit will have trouble reading 'null' path. It expects empty string if no config file is expected.
+  String optionsConfigFile = System.getProperty('beamITOptions', '')
+
+  String integrationTest = System.getProperty('integrationTest')
+  String itModule = System.getProperty('itModule')
+
+  String runner = System.getProperty('integrationTestRunner', 'direct')
+
+  String extraProperties = System.getProperty('beamExtraProperties', '')
+}
+
+// Configures a project with a set of tasks needed for running performance tests
+ext.enableJavaPerformanceTesting = {
+  println "enableJavaPerformanceTesting with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // Add runners needed to run integration tests on
+  task packageIntegrationTests(type: Jar) {
+    if (gradle.startParameter.taskNames.contains('integrationTest')) {
+
+      def runner = configuration.runner
+      dependencies {
+        if (runner.contains('dataflow')) {
+          testCompile project(path: ":runners:google-cloud-dataflow-java", configuration: 'shadowTest')
+        } else {
+          testCompile project(path: ":runners:direct-java", configuration: 'shadowTest')
+        }
+      }
+    }
+  }
+
+  // Task for running integration tests
+  task integrationTest(type: Test) {
+    include "**/*IT.class"
+    systemProperties.beamTestPipelineOptions = configuration.integrationTestPipelineOptions
+  }
+}
+
+ext.createPerformanceTestHarness = {
+  println "createPerformanceTestHarness with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // This task runs PerfKitBenchmarker, which does benchmarking of the IO ITs.
+  // The arguments passed to it allows it to invoke gradle again with the desired benchmark.
+  //
+  // To invoke this, run:
+  //
+  // ./gradlew performanceTest \
+  //  -DpkbLocation="<path to pkb.py>"
+  //  -DintegrationTestPipelineOptions='["--numberOfRecords=1000", "<more options>"]' \
+  //  -DintegrationTest=<io test, eg. org.apache.beam.sdk.io.text.TextIOIT> \
+  //  -DitModule=<directory containing desired test, eg. sdks/java/io/file-based-io-tests> \
+  //  -DintegrationTestRunner=direct/dataflow/<other>
+  //
+  // There are more options with default values that can be tweaked if needed (see below).
+  task performanceTest(type: Exec) {
+
+    // Perfkit needs to work in the Beam's root directory,
```

Comment:
You mean, test has to be invoked from the root directory ?

Reply:
I made this comment more clear in the next commit too. I meant that the performanceTest task requires perfkit to be run in the project root directory.